strings/base_events.h

```diff
@@ -375,8 +375,37 @@ WINRT_EXPORT namespace winrt
         using delegate_type = Delegate;
 
         event() = default;
-        event(event<Delegate> const&) = delete;
-        event<Delegate>& operator =(event<Delegate> const&) = delete;
+
+        event(event<Delegate> const& value)
```

Comment:
Making `event` copyable will result in previously-uncopyable types becoming copyable. I wonder if that will cause problems. (Maybe somebody had been relying on non-copyability of event to block some unwanted action. I confess that I may have done that.)

Reply:
Yes, I'm mainly concerned about this aspect of this PR. While it may be simple enough to do, I wonder whether this is worthwhile. Copying an event doesn't seem all that useful.

Reply:
I agree that in the general case it doesn't seem very useful, but in our use case the event is stored within a type that is captured by value in a lambda. The lambda then ends up stored in `std::function`, and `std::function` expects the callable type to be copyable. Wrapping the type holding the `event` or `event` itself in `std::unique_ptr` doesn't works, because it's move-only. While `std::shared_ptr` works, it adds ref-counting on top of existing ref-counting (from COM), so it adds some amount of overhead.

Reply:
That's more of a problem with `std::function` or whatever library that requires copies. The trouble is allowing copying may allow developers to easily introduce unexpected behavioral issues. So you allow copies, what does that mean? Are delegates expecting to be callable from multiple event sources? It just seems like this could open things up to some surprising issue.

Reply:
Right. This isn't a performance critical path anyways so wrapping it with std::shared_ptr will do for now. Will remove that from the PR and make it only clearable.

Reply:
Done